### DISCUSSION TITLE
cannot unmarshal string into Go value of type struct

### DISCUSSION THREAD

**Discussion by fenrirlobo** [qdm12/ddns-updater#562] [Q&A]
Hi guys,

The service is not working for me, its returning a error while reading the config.file.

Log end lines:
2023-11-27T17:09:23Z INFO reading JSON config from file /updater/data/config.json	json.go:L45
2023-11-27T17:09:23Z ERROR cannot unmarshal common settings: json: cannot unmarshal string into Go value of type struct { CommonSettings []params.commonSettings "json:\"settings\"" }	main.go:L70
2023-11-27T17:09:23Z INFO Shutdown successful	main.go:L84

Config file (hid just host subdomain, username and password):
{
    "settings": [
        {
            "provider": "dynu",
            "domain": "api.dynu.com",
            "host": "mysub.ddnsgeek.com",
            "username": "myusername",
            "password": "mypassword",
            "ip_version": "ipv4",
            "provider_ip": true
        }
    ]
}

Any idea of what am i doing wrong?
Thx in advance for any help!

**Comment by qdm12** [comment]
You are likely passing the entire json with surrounding quotes, so the json decoder thinks you are trying to decode a string and that doesn't match the programmed json scheme for decoding.

**Comment by fenrirlobo** [comment]
Hi,

So, thats is the wrong way (with quotes):
{
"settings": [
{
"provider": "dynu",
"domain": "api.dynu.com",
"host": "mysub.ddnsgeek.com",
"username": "myusername",
"password": "mypassword",
"ip_version": "ipv4",
"provider_ip": true
}
]
}

And thats the right way (without quotes):
{
settings: [
{
provider: dynu,
domain: api.dynu.com,
host: mysub.ddnsgeek.com,
username: myusername,
password: mypassword,
ip_version: ipv4,
provider_ip: true
}
]
}


?

Edit: I tried without any quotes but got the same error.

**Comment by fenrirlobo** [comment]
Edit 2: You'r right! ... the CasaOS editor messed it up... did the right way with nano and now its ok!
Tyvm!
![error](https://github.com/qdm12/ddns-updater/assets/139849448/bff801f7-777d-4d29-b37c-c348346a901c)
